### ISSUE TITLE
Shutdown Delay should deregister a nomad service

### ISSUE THREAD

**Issue by mr-karan** [hashicorp/nomad#15602]
### Nomad version
Output from `nomad version`

`Nomad v1.4.3 (f464aca721d222ae9c1f3df643b3c3aaa20e2da7)`

### Operating system and Environment details

Ubuntu 22.04

### Issue

The `group` in the below jobspec has a `shutdown_delay` param configured to be 30s.

According to the docs [here](https://developer.hashicorp.com/nomad/docs/job-specification/group) and [here](https://developer.hashicorp.com/nomad/docs/job-specification/task#shutdown_delay), only `consul` is configured to de-register the service.

> The delay occurs between Consul deregistration and sending each task a shutdown signal
> Specifies the duration to wait when killing a task between removing it from Consul and sending it a shutdown signal

If I infer this correctly, once the task is sent a SIGINT/SIGTERM, the service will get de-registered immediately however the actual task won't be killed until `shutdown_delay` param. Is my understanding correct here? 

In that case, the same behaviour doesn't happen with `nomad` service provider.

![image](https://user-images.githubusercontent.com/5689132/208872590-84374c25-5540-47b7-94ef-e6ff7fed1523.png)

### Reproduction steps

- Run the app: `nomad run deployment.nomad`
- Trigger a new deployment, since the jobspec has a `meta` variable with `uuidv4` value, re-running `nomad run deployment.nomad` would do the same.
- Watch the list of allocs and services and notice that the alloc which is given a `stop` signal, still is available in the registrations:

![image](https://user-images.githubusercontent.com/5689132/208872847-2dfe9719-df66-42ed-bed8-58546f2badf0.png)

#### Expected Result

The service should be de-registered immediately.

#### Actual Result

The service isn't removed until the alloc itself is stopped. Thus, configuring `shutdown_delay` isn't helping here.

### Job file (if appropriate)

```
job "doggo" {
  datacenters = ["dc1"]
  type        = "service"
  namespace   = "default"

  group "app" {
    count          = 3
    shutdown_delay = "30s"

    # Rolling update.
    update {
      max_parallel = 1
      stagger      = "15s"
    }


    network {
      port "http" {
        to = 8080
      }
    }

    task "web" {
      driver = "docker"

      service {
        provider = "nomad"
        name     = "doggo-web"
        tags     = ["doggo", "web"]
        port     = "http"
        check {
          type     = "http"
          port     = "http"
          path     = "/api/ping/"
          interval = "5s"
          timeout  = "2s"
        }
      }

      meta {
        deployment_id = uuidv4()
      }

      config {
        image   = "ghcr.io/mr-karan/doggo-api:latest"
        ports   = ["http"]
        command = "/bin/sh"
        args    = ["-c", "sleep 10 && /app/doggo-api.bin"]
      }

      resources {
        cpu    = 200
        memory = 200
      }
    }
  }
}
```

**Comment by mr-karan** [comment]
This definitely makes sense. Thanks @jrasell for the help :) Should I keep the issue open if this warrants a documentation update?

**Comment by jrasell** [comment]
Yeah let's keep it open; I will look at the documentation tomorrow and close this once they have been updated.
